src/preset.ts

```diff
@@ -392,28 +504,42 @@ function getVendorForConfigurePresetHelper(folder: string, preset: ConfigurePres
   return preset.vendor || null;
 }
 
-export async function expandConfigurePreset(folder: string,
-                                            name: string,
-                                            workspaceFolder: string,
-                                            sourceDir: string,
-                                            allowUserPreset: boolean = false): Promise<ConfigurePreset | null> {
-  const refs = referencedConfigurePresets.get(folder);
-  if (!refs) {
-    referencedConfigurePresets.set(folder, new Set());
+async function getHostSystemName(): Promise<string> {
```

Comment:
Nit: you may want to put this to util.ts since this doesn't require anything in this file. Also, you can use memento in VS Code (`extensionContext.globalState`) to store the value so this value can be persisted in all VS Code instances.

Reply:
Thanks, I was wondering if VSCode had a built in way to handle that. Two questions though:
1. Does `extensionContext.globalState` store state independently for each platform? From [the docs](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.globalState) the storage is independent of any workspace but it doesn't mention the behavior if using the remote extensions. My guess is it is independent (my understanding is when using the remote extension the headless server process is an independent install, but I'm not sure if that's correct).
2. Is the only way to get a reference to `extensionContext` through `CMakeTools`? If so it means we'd have to pass that through the expand*Preset functions, which seems awkward to me.

Reply:
1. It's independent, since extension is installed separately on the remote machine in remote case (for this extension).
2. That's a good questions. Actually you probably don't want to use CMakeTools in this file since it depends on this file. Then I guess the current way is probably good enough. And still, I don't have a strong preference on this but you may want move this to util.ts.

Reply:
Thanks for confirming. I'll move it to util.ts but I'll keep the current memo code to avoid passing in CMakeTools